src/test/java/eu/europeana/metis/sandbox/service/dataset/DatasetGeneratorServiceImplTest.java

```diff
@@ -60,18 +71,18 @@ void generate_emptyRecords_expectFail() {
     assertThrows(IllegalArgumentException.class, () -> generator.generate(getTestDatasetMetadata(), List.of()));
   }
 
-  @Test
-  void generate_inCaseOfInvalidRecords_expectSuccess() {
-    when(xmlRecordProcessorService.getRecordId("record1".getBytes())).thenThrow(RecordParsingException.class);
-    when(xmlRecordProcessorService.getRecordId("record2".getBytes())).thenReturn("2");
-    when(xmlRecordProcessorService.getRecordId("record3".getBytes())).thenThrow(RecordParsingException.class);
-    when(xmlRecordProcessorService.getRecordId("record4".getBytes())).thenThrow(IllegalArgumentException.class);
-    when(xmlRecordProcessorService.getRecordId("record5".getBytes())).thenReturn("5");
-
-    Dataset dataset = generator.generate(getTestDatasetMetadata(), getTestRecords());
-
-    assertEquals(2, dataset.getRecords().size());
-  }
```

Comment:
Please, see my comment on ```DatasetGeneratorServiceImpl```, you this unit test needs to refactor, to keep safe guarding MET-4093 when an exception is thrown by the ```repository.save``` method.

Reply:
Ooops, completely missed this! Thanks!
I changed the exceptions types you wrote to the ones that were originally in the test, since those exceptions are "expected" to be thrown. Check my comment here https://github.com/europeana/metis-sandbox/pull/44/files#r802591491